codingstandards/listproperties.md

```diff
@@ -0,0 +1,128 @@
+# List Properties
+
+Many types, especially data structures, will contain properties of a list type,
+such as `IList<T>` or `IReadOnlyList<T>`.  List properties should be implemented
+under the following guidelines, especially `public` properties.
+
+- The list property itself should be read-only.
+- The list property should be initialized to a new instance when the containing data structure is constructed.
+- The list property should be of an interface type when possible.  That is, use `IList<T>` rather than `List<T>`.
+- Use the read-only collections when appropriate, such as `IReadOnlyCollection<T>`, `IReadOnlyList<T>`, and `IReadOnlyDictionary<TKey,TValue>`.
+
+## Validation
+
+As with any type, validation of the input should be performed as early as possible.
+In most cases, this can be accomplished in the constructor of the class.
+In the cases of lists, however, when a list is expected to only contain certain values,
+validation needs to occur as values are added, inserted, or set using an indexer.
+
+Here is an example of how to **not** validate the entries of a list.
+Note the list property, `Attributes`, correctly defined as a read-only property as `IList<string>`.
+
+```csharp
+public class Example
+{
+   public string Property { get; }
```

Comment:
@johntkeller For consistency with how we expect these to be, I think we should add the `[DataContract]` attribute to this to instill memorization. I know it is sort of unrelated, but it might be confusing for developers if we are using it in some cases and not others when in reality we should be applying it in all cases.